source/upgrading/upgrade/upgrade-4.13.rst

```diff
@@ -0,0 +1,360 @@
+each system with CloudStack packages. This means all management
+servers, and any hosts that have the KVM agent. (No changes should
+be necessary for hosts that are running VMware or Xen.)
+
+
+
+Start by opening ``/etc/apt/sources.list.d/cloudstack.list`` on
+any systems that have CloudStack packages installed.
+
+This file should have one line, which contains:
+
+.. parsed-literal::
+
+   deb http://download.cloudstack.org/ubuntu precise 4.13
+
+We'll change it to point to the new package repository:
+
+.. parsed-literal::
+
+   deb http://download.cloudstack.org/ubuntu precise |version|
+
+Setup the public key for the above repository:
+
+.. parsed-literal::
+
+   wget -qO - http://download.cloudstack.org/release.asc | sudo apt-key add -
+
+If you're using your own package repository, change this line to
+read as appropriate for your |version| repository.
+
+#. Now update your apt package list:
+
+   .. parsed-literal::
+
+      $ sudo apt-get update
+
+#. Now that you have the repository configured, it's time to upgrade
+   the ``cloudstack-management`` package.
+
+   .. parsed-literal::
+
+      $ sudo apt-get upgrade cloudstack-management
+
+#. If you use CloudStack usage server
+
+   .. parsed-literal::
+
+      $ sudo apt-get upgrade cloudstack-usage
+
+
+.. _rhel413:
+.. _rpm-repo413:
+
+CentOS/RHEL
+##############
+
+If you are using CentOS or RHEL, follow this procedure to upgrade your
+packages. If not, skip to hypervisors section :ref:`upg_hyp_413`.
+
+.. note::
+   **Community Packages:** This section assumes you're using the community
+   supplied packages for CloudStack. If you've created your own packages and
+   yum repository, substitute your own URL for the ones used in these examples.
+
+The first order of business will be to change the yum repository
+for each system with CloudStack packages. This means all
+management servers, and any hosts that have the KVM agent.
+
+(No changes should be necessary for hosts that are running VMware
+or Xen.)
+
+Start by opening ``/etc/yum.repos.d/cloudstack.repo`` on any
+systems that have CloudStack packages installed.
+
+This file should have content similar to the following:
+
+.. parsed-literal::
+
+   [apache-cloudstack]
+   name=Apache CloudStack
+   baseurl=http://download.cloudstack.org/centos/7/4.13/
+   enabled=1
+   gpgcheck=0
+
+If you are using the community provided package repository, change
+the base url to:
+
+.. parsed-literal::
+
+   http://download.cloudstack.org/centos/$releasever/|version|/
+
+Setup the GPG public key if you wish to enable ``gpgcheck=1``:
+
+.. parsed-literal::
+
+   rpm --import http://download.cloudstack.org/RPM-GPG-KEY
+
+
+
+If you're using your own package repository, change this line to
+read as appropriate for your |version| repository.
+
+#. Now that you have the repository configured, it's time to upgrade the
+   ``cloudstack-management``.
+
+   .. parsed-literal::
+
+      $ sudo yum upgrade cloudstack-management
```

Comment:
if not already installed this will install a jdk 11 version but it will not set it as the default. so anybody who hasn't already installed the new jdk needs an extra step; `alternatives --config java` and choose the jdk/jre containing the right jvm version.

Reply:
I'll add a line.